### PR TITLE
feat: migrate WingDex from GitHub Spark to Cloudflare Pages + D1

### LINKED ISSUE #74: Migrate off of GitHub Spark?
Spark was nice for initial bootstrapping but quickly had to switch to regular Codespaces based development. Which has been ok except the LLM calls (although included and free) are only allowed in the specific codespaces, the codespaces themselves have some weird quirks, deployment to prod is slightly mysterious, no custom domain, weird favicon-adjacent behaviors, etc. 

But then we'd need to find a (free or cheap?) provider that can host the server side Node app, KV store, authN, LLM calls (maybe GitHub models?). Cloudflare has functions and KV? Azure free tier?

### PR DESCRIPTION
**Pull request by jlian**
## Summary

Complete migration of WingDex from GitHub Spark to Cloudflare Pages + D1 + Better Auth + Workers AI. Resolves #74.

Verified on `dev.wingdex.pages.dev` — passkey auth, GitHub/Apple OAuth sign-in, account linking, data persistence, bird ID, eBird import/export all working.

### What changed

**Platform migration (Phases 0-4):**
- Replaced Spark auth (`window.spark.user()`) with Better Auth (passkey-first + anonymous bootstrap signup flow)
- Replaced Spark KV persistence with D1/SQLite via relational schema (outings, photos, observations, dex_meta)
- Replaced Spark LLM proxy with server-owned `/api/identify-bird` endpoint (OpenAI via AI Gateway)
- Removed Spark runtime bootstrap, Vite plugins, config files, and `@github/spark` dependency
- Added `wrangler.toml`, D1 migrations (4 files), and `functions/` directory with Pages Functions

**Auth (Phase 1):**
- Passkey-first signup via anonymous bootstrap -> addPasskey -> finalize (dialog UX with random bird-name generator)
- GitHub + Apple OAuth sign-in + account linking (`allowDifferentEmails: true`)
- "Link account" buttons in Settings (conditional on env var + not already linked)
- Two OAuth environments: prod (`wingdex.app`) and dev (`dev.wingdex.pages.dev`)
- Apple client secret auto-rotation via GitHub Actions workflow

**Server-side business logic (Phases 2-3):**
- Moved bird ID pipeline, taxonomy search, eBird CSV parsing, and dex aggregation to server
- API-first design: `/api/data/*`, `/api/import/*`, `/api/export/*`, `/api/species/*`, `/api/identify-bird`
- Per-user daily AI rate limiting via D1
- Client bundle reduced ~876KB by removing `taxonomy.json` from SPA

**Testing (Phase 5):**
- 511 tests across 31 files, all passing
- Added server-side function tests (taxonomy, eBird, bird-id prompt/helpers, rate limiting, dex query)
- E2E: 33 passed / 1 skipped (known anonymous-auth edge case)
- TypeScript 0 errors, ESLint 0 warnings

**CI/CD and Deploy (Phase 6):**
- `release.yml`: push to main/dev -> semantic-release (main only) -> D1 migrations -> Cloudflare Pages deploy
- `ci.yml`: PR quality gate (lint -> typecheck -> unit -> build -> migrate local D1 -> e2e -> preview deploy)
- Separate D1 databases: `wingdex-db` (prod) and `wingdex-db-dev` (preview)
- `node_modules` + Playwright browser caching for faster CI
- GitHub + Apple + Cloudflare secrets configured for both environments
- Custom domain `wingdex.app` added (pending first production deploy)

**Cleanup (Phase 7):**
- Removed dead code: `textLLM`, `/api/suggest-location`, unused exports
- Replaced Radix Select with native `<select>` (chunk 113KB -> 63KB)
- Updated PRD, README, PWA manifests to Cloudflare/Better Auth terminology
- Improved legal pages (privacy, terms) with full React component text
- Removed stale `npm@10.9.2` version pin
- Fixed `computeFileHash` for files <128KB

### Stats

- 136 files changed, ~13K insertions, ~4.5K deletions
- 143 commits

### Remaining work (post-merge)

- Google OAuth app registration (Phase 1.12)
- First production deploy to `wingdex.app`

### CODE REVIEW

File: functions/api/data/photos.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  const uniqueOutingIds = Array.from(new Set(outingIds))
+  if (uniqueOutingIds.length === 0) return true
+
+  const placeholders = uniqueOutingIds.map(() => '?').join(', ')
+  const result = await db
+    .prepare(`SELECT id FROM outing WHERE userId = ? AND id IN (${placeholders})`)
+    .bind(userId, ...uniqueOutingIds)
+    .all<{ id: string }>()
+
+  return result.results.length === uniqueOutingIds.length
```
[the remark this answers is not in the document]
Comment (jlian) [reply functions/api/data/photos.ts]: The empty array case is already guarded at line 27-28: `if (uniqueOutingIds.length === 0) return true`. The `IN ()` scenario cannot occur. Re: D1 bind parameter limits — the function is only called with outing IDs from a single photo upload batch, which is capped at a small number by the client. Not a concern in practice.

File: functions/lib/ai-rate-limit.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    .run()
+
+  const increment = await db.prepare(
+    `UPDATE ai_daily_usage
+       SET requestCount = requestCount + 1,
+           updatedAt = datetime('now')
+     WHERE userId = ? AND endpoint = ? AND usageDate = ? AND requestCount < ?`,
+  )
+    .bind(userId, endpoint, today, limit)
+    .run()
```
[the remark this answers is not in the document]
Comment (jlian) [reply functions/lib/ai-rate-limit.ts]: Acknowledged. The race window is between `INSERT … ON CONFLICT DO NOTHING` and `UPDATE … WHERE requestCount < limit`. In practice, the limit is 150 requests/day for bird identification — a ±1 overshoot at the boundary is completely acceptable for this use case. D1 does not support `SELECT FOR UPDATE`, and a single atomic conditional increment would require a more complex upsert. Not worth the complexity for a soft daily cap.

File: wrangler.toml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+name = "wingdex"
+compatibility_date = "2025-11-01"
```
[the remark this answers is not in the document]
Comment (jlian) [reply wrangler.toml]: Please learn to understand time and dates better.

**Comment by jlian** [comment]
I think the footer is a bit too low and too wide on iOS

image

Maybe make it two lines like GitHub?

image

File: src/components/flows/AddPhotosFlow.tsx [hunk 4]
```diff
@@ -248,8 +253,8 @@ export default function AddPhotosFlow({ data, onClose, userId }: AddPhotosFlowPr
       if (confirmed.length > 0) {
         toast.success(`All done! ${confirmed.length} species saved.`)
       }
-      setStep('complete')
-      setTimeout(onClose, 3500)
+      setShowConfetti(false)
+      onClose()
     }
```
[the remark this answers is not in the document]
Comment (jlian) [reply src/components/flows/AddPhotosFlow.tsx]: Addressed in b6cce64: removed the ineffective confetti state/render path in AddPhotosFlow because the final-cluster completion path immediately closes the dialog; completion feedback remains via toast.

[the remark this answers is not in the document]
Comment (jlian) [reply src/components/flows/AddPhotosFlow.tsx]: Addressed in b6cce64: removed the ineffective confetti state/render path in AddPhotosFlow because the final-cluster completion path immediately closes the dialog; completion feedback remains via toast.

File: src/lib/fun-names.ts [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+}
+
+/** Return the Tailwind color class for an emoji avatar data-URL, or '' if not an emoji avatar. */
+export function getEmojiAvatarColor(imageUrl: string | undefined | null): string {
+  if (!imageUrl?.startsWith('data:image/svg+xml')) return ''
+  const decoded = decodeURIComponent(imageUrl)
+  for (const [emoji, color] of Object.entries(emojiColors)) {
+    if (decoded.includes(emoji)) return color
+  }
+  return ''
```
[the remark this answers is not in the document]
Comment (jlian) [reply src/lib/fun-names.ts]: Addressed in 91deb9c: wrapped decodeURIComponent in getEmojiAvatarColor with try/catch and return '' on decode failure to avoid render-time exceptions from malformed/non-encoded data URIs.
